docs/index.rst

```diff
@@ -106,24 +106,80 @@ You can use pip and make to quickly set up a development environment::
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to fix them.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
-
+* The output of entity operations is a must, either use Satellite or (Foreman+Katello). Outputs should be provided for every new/affected helper or field in the entity. Or, if you have some tests exercising corresponding fields/helpers (for robottelo or other repo) - providing just test results is enough. This should make the review process easier for all reviewers and anyone else interested in the new functionality.
+* When raising a PR against any other branch which is also applicable for '6.X' or foreman branch. Please specify that information in the PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
 * When in doubt, ask on IRC.
 
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    The moment PR would be raised, contributor needs to set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added  while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm. The PR author can provide results by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
+
+Nailgun Review Process
+______________________
+
+* Travis CI is run, and any issues are resolved by contributor.
+* If deemed necessary by contributors/reviewers, an automation run is triggered.
+* At least two ACKs are required to merge a pull request.
+* At least one ACK must be from a Tier 2 reviewer.
```

Comment:
Can we define somewhere what groups of people are a Tier 1/2 reviewer?

Reply:
@sean797 Fully agreed with you. In the meantime, until this is resolved, you might be interested in userjs script I created - https://github.com/SatelliteQE/userscripts . It adds list of Tier1 and Tier2 reviewers near "Reviewers" on top right. Right now, list contains only people who were identified for each tier internally in SatelliteQE. As Nailgun in community project, it is not exhaustive here. 

And yes, I do expect to find you on that list once we deal with formalities :) .